### PR TITLE
[FORUM-1052]Category spaces appears in the list when creating a forum

### PR DESCRIPTION
**Pull request by AtiaZ**
Fix description : 

1- Don't display the Category "Spaces" in the Forum Portlet.
2- remove the "Spaces" category from the select field when creating a new forum.

### CODE REVIEW

File: forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java [hunk 1]
```diff
@@ -321,7 +320,25 @@ public Category getCategoryIncludedSpace() {
...
   public List<Category> getCategories() {
-    return storage.getCategories();
+    return getCategories(true);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public List<Category> getCategories(boolean includeSpacesCategories) {
+    List<Category> DisplayedCategories = new ArrayList<Category>();
```
Comment (thomasdelhomenie) [inline forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java]: DisplayedCategories -> displayedCategories

File: forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java [hunk 2]
```diff
@@ -321,7 +320,25 @@ public Category getCategoryIncludedSpace() {
...
+    List<Category> categories = storage.getCategories();
+    if (includeSpacesCategories) {
+      DisplayedCategories = categories;
+    } else {
+      for (Category category : categories) {
+        if (!(category.getId().equals("forumCategoryspaces"))) {
```
Comment (thomasdelhomenie) [inline forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java]: We should have a constant for "forumCategoryspaces"

File: forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java (hunk 1 above)
Comment (AtiaZ) [inline forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java]: done

File: forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java (hunk 2 above)
Comment (AtiaZ) [inline forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java]: done

File: forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java [hunk 3]
```diff
@@ -101,6 +100,8 @@
   private UserStateService           userStateService;
 
   protected List<ForumEventListener> listeners_      = new ArrayList<ForumEventListener>(3);
+
+  private static final String FORUM_CATEGORY_SPACES = "forumCategoryspaces";
```
Comment (thomasdelhomenie) [inline forum/service/src/main/java/org/exoplatform/forum/service/impl/ForumServiceImpl.java]: Actually we already have a constant for this value : org.exoplatform.forum.service.Utils.CATEGORY_SPACE_ID_PREFIX

File: forum/service/src/test/java/org/exoplatform/forum/service/ForumServiceTestCase.java [hunk 4]
```diff
@@ -58,6 +57,60 @@ public void tearDown() throws Exception {
     removeAllData();
     super.tearDown();
   }
+
+  public void testCategoriesListWithoutSpacesCategory() throws Exception {
+    // Given
+    String cateId = Utils.CATEGORY + "spaces";
```
Comment (thomasdelhomenie) [inline forum/service/src/test/java/org/exoplatform/forum/service/ForumServiceTestCase.java]: Same here, use Utils.CATEGORY_SPACE_ID_PREFIX instead.

**Comment by AtiaZ** [comment]
Merged

**Pull request closed without merging**